transport/http/config.go

```diff
@@ -154,22 +152,38 @@ type OutboundConfig struct {
 	// URL to which requests will be sent for this outbound. This field is
 	// required.
 	URL string `config:"url,interpolate"`
+
+	// HTTP headers that will be sent with all requests made through this
+	// outbound.
+	//
+	//  http:
+	//    url: "http://localhost:8080/yarpc"
+	//    headers:
+	//      X-Caller: myserice
+	//      X-Token: foo
+	Headers map[string]string `config:"headers"`
```

Comment:
Maybe we should call it `ExtraHeaders`. Just to make it clear that they are extra headers added to every request? I don't have a strong opinion on this though so feel free to keep it this way.